### ISSUE TITLE
Console application boots Silex application

### ISSUE THREAD

**Issue by sokil** [KnpLabs/ConsoleServiceProvider#15]
ConsoleServiceProvider must be registered last, because it boots silex application providers.

I want to use `\Kurl\Silex\Provider\DoctrineMigrationsProvider` for migrations. This provider require Console application, so i add it:

``` php
<?php
$app->register(new ConsoleServiceProvider(), array(
    'console.name'              => 'app',
    'console.version'           => '1.0.0',
    'console.project_directory' => __DIR__
));

// at this moment all previders registered in silex app are booted

$app->register(
    new \Kurl\Silex\Provider\DoctrineMigrationsProvider($app['console']),
    array(
        'migrations.directory'  => __DIR__ . '/../migrations',
        'migrations.name'       => 'app',
        'migrations.namespace'  => 'App\Migrations',
        'migrations.table_name' => 'migrations',
    )
);

// at this moment to late to register anything. Application already booted
```

**Comment by satahippy** [comment]
+1

**Comment by imelectronick** [comment]
+1

**Comment by IMM0rtalis** [comment]
+1 came here to report this too! 
What's the status?

**Comment by skalpa** [comment]
Fix merged in #32.

People who depend on the old behavior can set the new `console.boot_in_constructor` parameter to `true`.
